tests/drum/test_units.py

```diff
@@ -726,6 +766,26 @@ def test_yaml_metadata__multi_hyper_param_optional_fields(
         assert optional_field not in model_metadata[ModelMetadataKeys.HYPERPARAMETERS][0]
 
 
+@pytest.mark.parametrize(
+    "optional_component_param_keys",
+    itertools.combinations(ModelMetadataMultiHyperParamTypes.all(), 1),
+)
+def test_yaml_metadata__multi_hyper_param_optional_component_params(
+    optional_component_param_keys, complete_multi_hyper_param, tmp_path, basic_model_metadata_yaml,
+):
```

Comment:
Can you add a docstring or some comments on this test

Reply:
https://github.com/datarobot/datarobot-user-models/blob/e6bdbed5a969312cbb201ad0d48d6c804a26beee/tests/drum/test_units.py#L770-L777